client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-recipe.ts

```diff
@@ -0,0 +1,130 @@
+import { useColorPaletteVariations, useFontPairingVariations } from '@automattic/global-styles';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState, useEffect, useRef, useCallback } from 'react';
+import { ONBOARD_STORE } from '../../../../../stores';
+import type { Pattern } from '../types';
+import type { OnboardSelect } from '@automattic/data-stores';
+import type { Design } from '@automattic/design-picker/src/types';
+import type { GlobalStylesObject } from '@automattic/global-styles';
+
+const useRecipe = ( siteId = 0, patterns: Pattern[] ) => {
+	const incrementIndexRef = useRef( 0 );
+	const [ header, setHeader ] = useState< Pattern | null >( null );
+	const [ footer, setFooter ] = useState< Pattern | null >( null );
+	const [ sections, setSections ] = useState< Pattern[] >( [] );
+	const [ colorVariation, setColorVariation ] = useState< GlobalStylesObject | null >( null );
+	const [ fontVariation, setFontVariation ] = useState< GlobalStylesObject | null >( null );
+	const selectedDesign = useSelect(
+		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getSelectedDesign(),
+		[]
+	);
+
+	const {
+		stylesheet = '',
+		header_pattern_ids = [],
+		footer_pattern_ids = [],
+		pattern_ids = [],
+		color_variation_title = '',
+		font_variation_title = '',
+	} = selectedDesign?.recipe || {};
+
+	const colorVariations = useColorPaletteVariations( siteId, stylesheet );
+
+	const fontVariations = useFontPairingVariations( siteId, stylesheet );
```

Comment:
We have to load colors & fonts here as they will be loaded only when the user goes to the colors or fonts screen. We can check the values of `color_variation_title` and `font_variation_title` to determine whether to load the data. But the endpoint is fast enough so I think it's fine to load the variations at the begining.